### ISSUE TITLE
Wrong diagnostic label text due to text replacement

### ISSUE THREAD

**Issue by ytausky** [rust-lang/rust#51244]
The following snippet:
```rust
fn main() {
    let ref my_ref @ _ = 0;
    *my_ref = 0
}
```
Produces the following output:
```
error[E0594]: cannot assign to immutable borrowed content `*my_ref`
 --> src/main.rs:3:5
  |
2 |     let ref my_ref @ _ = 0;
  |         -------------- consider changing this to `ref mut my_ref mut @ _`
3 |     *my_ref = 0
  |     ^^^^^^^^^^^ cannot borrow as mutable
```
Note that the label suggests invalid syntax. I believe this happens due to the text replacement in the following code:
https://github.com/rust-lang/rust/blob/e38554cd80af2e99d626230ad3531b55ee3de65c/src/librustc_borrowck/borrowck/mod.rs#L1210-L1219
